common/pupil/screening.ts

```diff
@@ -27,3 +28,23 @@ export async function addPupilScreening(pupil: Pupil, screening: PupilScreeningI
 
     logger.info(`Added ${screening.status || 'pending'} screening for pupil ${pupil.id}`, screening);
 }
+
+interface PupilScreeningUpdate {
+    status?: PupilScreeningStatus;
+    comment?: string;
+}
+
+export async function updatePupilScreening(pupilScreeningId: number, screeningUpdate: PupilScreeningUpdate) {
+    const screening = await prisma.pupil_screening.findFirst({ where: { id: pupilScreeningId }, include: { pupil: {} } });
+    if (screening === null) {
+        logger.error(`cannot find PupilScreening(${pupilScreeningId})`);
+        throw new NotFoundError('pupil screening not found');
+    }
+
+    await prisma.pupil_screening.update({ where: { id: pupilScreeningId }, data: { ...screeningUpdate, updatedAt: new Date() } });
+    logger.debug(`successfully updated PupilScreening(${pupilScreeningId})`, { pupilScreeningId, screeningUpdate });
+
+    if (screeningUpdate.status === PupilScreeningStatus.success || screeningUpdate.status === PupilScreeningStatus.rejection) {
```

Comment:
And I guess this should also check that the previous status was pending, otherwise editing multiple times might accidentally trigger multiple notifications.

Reply:
Good catch